CC/src/DistanceComputationTools.cpp

```diff
@@ -2464,6 +2465,144 @@ int DistanceComputationTools::computeCloud2PlaneEquation(GenericIndexedCloudPers
 	return count;
 }
 
+int DistanceComputationTools::computeCloud2RectangleEquation(GenericIndexedCloudPersist *cloud, const PointCoordinateType* planeEquation, const PointCoordinateType widthX, const PointCoordinateType widthY, const CCVector3 center, CCVector3 xColumnGL, CCVector3 yColumnGL, CCVector3 zColumnGL) {
+	// p3---------------------p2
+	// ^					  |
+	// |e1					  |
+	// |					  |
+	// |		  e0		  |
+	// p0-------------------->p1
+	//Rect(s,t)=p0 + s*0e + t*e1 
+	//for s,t in {0,1}
+	SquareMatrix rotationTransform = SquareMatrix::SquareMatrixTpl(3);
+	rotationTransform.setValue(0, 0, xColumnGL[0]);
+	rotationTransform.setValue(1, 0, xColumnGL[1]);
+	rotationTransform.setValue(2, 0, xColumnGL[2]);
+	rotationTransform.setValue(0, 1, yColumnGL[0]);
+	rotationTransform.setValue(1, 1, yColumnGL[1]);
+	rotationTransform.setValue(2, 1, yColumnGL[2]);
+	rotationTransform.setValue(0, 2, zColumnGL[0]);
+	rotationTransform.setValue(1, 2, zColumnGL[1]);
+	rotationTransform.setValue(2, 2, zColumnGL[2]);
+	float d = static_cast<float>(widthX);
```

Comment:
Prefer 'PointCoordinateType' over 'float' (for the 'double' version of CloudCompare). Or if you really want float values here, you should make this explicit (with static_cast) to avoid warnings.